lib/gym.rb

```diff
@@ -35,7 +30,34 @@ def gymfile_name
     def xcode_path
       @path ||= `xcode-select --print-path`.strip
     end
+
+    def xcode_version
+      @version ||= parse_version
+    end
+
+    def pre_7?
+      v = xcode_version
+      is_pre = v.split('.')[0].to_i < 7
+      is_pre
+    end
+
+    def init_libs
+      return unless pre_7?
+      # Import all the fixes
+      require 'gym/xcodebuild_fixes/swift_fix'
+      require 'gym/xcodebuild_fixes/watchkit_fix'
+      require 'gym/xcodebuild_fixes/package_application_fix'
+    end
+
+    private
+
+    def parse_version
+      output = `DEVELOPER_DIR='' "#{xcode_path}/usr/bin/xcodebuild" -version`
+      return '0.0' if output.nil?
+      output.split("\n").first.split(' ')[1]
+    end
   end
 
   Helper = FastlaneCore::Helper # you gotta love Ruby: Helper.* should use the Helper class contained in FastlaneCore
+  Gym.init_libs
```

Comment:
I don't like it either. A better idea ?
Do you prefer me to rewrite init_lib to be idempotent if needed (maybe it is already, not really sure what happens if one calls require multiple times), and then we just call it from all xcode < 7 codepaths ?